primitives/arithmetic/src/fixed_point.rs

```diff
@@ -400,7 +409,7 @@ macro_rules! implement_fixed {
 			type Output = Self;
 
 			fn neg(self) -> Self::Output {
-				Self(-self.0)
+				Self(<Self as FixedPointNumber>::Inner::zero() - self.0)
```

Comment:
What's the separation of `negate` and this `neg` impl?

Reply:
`neg` should panics when `self = min_value()` instead of saturate.